dashboard/test/factories/factories.rb

```diff
@@ -628,110 +628,95 @@
   factory :school_info_without_country, class: SchoolInfo do
     school_type SchoolInfo::SCHOOL_TYPE_PUBLIC
     state 'WA'
-    association :school_district
+    association :school_district, strategy: :build
```

Comment:
What is the difference between `association :school_district, strategy: :build` and `school_district: {build :school_district}`? We seem to use a mix of both patterns.

Reply:
The first is an [association](http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Associations), a reference to another factory. The second is a [dynamic attribute](http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Dynamic_Attributes), which is a block of lazily executed code. The association specified with `strategy: :build` will match the method (build/create) of the parent, whereas the dynamic attribute is just arbitrary code. Associations are preferable in most cases, but [here](https://github.com/code-dot-org/code-dot-org/pull/20852/files#diff-286a7ecb55a740b99e7f8b0fb50a8b8dR650) I chose the latter so I can refer to previously defined attributes. Associations do not have a dynamic component.